packages/manager/modules/pci/src/projects/project/serving/namespace/models/model.class.js

```diff
@@ -0,0 +1,34 @@
+
+import { PROCESSING_VERSION_STATUS, ERROR_VERSION_STATUS } from './models.constants';
+
+export default class Model {
+  constructor({
+    apiStatus,
+    createdAt,
+    id,
+    replicas,
+    storagePath,
+    url,
+    version,
+    versionStatus,
+  }) {
+    Object.assign(this, {
+      apiStatus,
+      createdAt,
+      id,
+      replicas,
+      storagePath,
+      url,
+      version,
+      versionStatus,
+    });
+  }
+
+  versionError() {
```

Comment:
Imo, functions name should contain verbs because they represent either an action or a check/state. Something like 
```suggestion
 isInError() {
```
might be more appropriate

Reply:
9eb0962dad0527f7e808bd6b11ebe3c3cfec136e